src/display/editor/highlight.js

```diff
@@ -0,0 +1,491 @@
+
+    // The last point is in the pages coordinate system.
+    const { lastPoint } = this.#focusOutlines.box;
+    this.#lastPoint = [
+      (lastPoint[0] - this.x) / this.width,
+      (lastPoint[1] - this.y) / this.height,
+    ];
+  }
+
+  static initialize(l10n) {
+    AnnotationEditor.initialize(l10n);
+  }
+
+  static updateDefaultParams(type, value) {
+    switch (type) {
+      case AnnotationEditorParamsType.HIGHLIGHT_COLOR:
+        HighlightEditor._defaultColor = value;
+        break;
+      case AnnotationEditorParamsType.HIGHLIGHT_OPACITY:
+        HighlightEditor._defaultOpacity = value / 100;
+        break;
+    }
+  }
+
+  /** @inheritdoc */
+  get toolbarPosition() {
+    return this.#lastPoint;
+  }
+
+  /** @inheritdoc */
+  updateParams(type, value) {
+    switch (type) {
+      case AnnotationEditorParamsType.HIGHLIGHT_COLOR:
+        this.#updateColor(value);
+        break;
+      case AnnotationEditorParamsType.HIGHLIGHT_OPACITY:
+        this.#updateOpacity(value);
+        break;
+    }
+  }
+
+  static get defaultPropertiesToUpdate() {
+    return [
+      [
+        AnnotationEditorParamsType.HIGHLIGHT_COLOR,
+        HighlightEditor._defaultColor,
+      ],
+      [
+        AnnotationEditorParamsType.HIGHLIGHT_OPACITY,
+        Math.round(HighlightEditor._defaultOpacity * 100),
+      ],
+    ];
+  }
+
+  /** @inheritdoc */
+  get propertiesToUpdate() {
+    return [
+      [
+        AnnotationEditorParamsType.HIGHLIGHT_COLOR,
+        this.#color || HighlightEditor._defaultColor,
+      ],
+      [
+        AnnotationEditorParamsType.HIGHLIGHT_OPACITY,
+        Math.round(100 * (this.#opacity ?? HighlightEditor._defaultOpacity)),
+      ],
+    ];
+  }
+
+  /**
+   * Update the color and make this action undoable.
+   * @param {string} color
+   */
+  #updateColor(color) {
+    const savedColor = this.color;
+    this.addCommands({
+      cmd: () => {
+        this.#color = color;
+        this.parent.drawLayer.changeColor(this.#id, color);
+      },
+      undo: () => {
+        this.#color = savedColor;
+        this.parent.drawLayer.changeColor(this.#id, savedColor);
+      },
+      mustExec: true,
+      type: AnnotationEditorParamsType.HIGHLIGHT_COLOR,
+      overwriteIfSameType: true,
+      keepUndo: true,
+    });
+  }
+
+  /**
+   * Update the opacity and make this action undoable.
+   * @param {number} opacity
+   */
+  #updateOpacity(opacity) {
+    opacity /= 100;
+    const savedOpacity = this.#opacity;
+    this.addCommands({
+      cmd: () => {
+        this.#opacity = opacity;
+        this.parent.drawLayer.changeOpacity(this.#id, opacity);
+      },
+      undo: () => {
+        this.#opacity = savedOpacity;
+        this.parent.drawLayer.changeOpacity(this.#id, savedOpacity);
+      },
+      mustExec: true,
+      type: AnnotationEditorParamsType.HIGHLIGHT_OPACITY,
+      overwriteIfSameType: true,
+      keepUndo: true,
+    });
+  }
+
+  /** @inheritdoc */
+  fixAndSetPosition() {
```

Comment:
Given that this looks quite similar to the base-class method, with `rotation === 0`, would it work and if so make sense to change https://github.com/mozilla/pdf.js/blob/b7e943bc7d31ef4e8f19c60a02eb9e70c2d17eb7/src/display/editor/editor.js#L510 into e.g. `fixAndSetPosition(forceRotation = null) {` and then modify https://github.com/mozilla/pdf.js/blob/b7e943bc7d31ef4e8f19c60a02eb9e70c2d17eb7/src/display/editor/editor.js#L518 to `switch (forceRotation ?? this.rotation) {` instead?